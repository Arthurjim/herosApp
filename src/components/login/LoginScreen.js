import React, { useContext } from "react";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { types } from "../../types/types";
import useForm from "../customHooks/useForm";
const LoginScreen = () => {
    const { dispatch } = useContext(AuthContext);
    const navigate = useNavigate();
    const [inputValue,handleInput] = useForm({'username':''})
    const handleLogin = () => {
        const action = {
            type: types.login,
            payload: {
                name: inputValue.username,
            },
        };
        dispatch(action);
        const path = localStorage.getItem('lastPath') || '/';
        navigate(path, { replace: true });
    };
    return (
        <div className="container mt-5">
            <h1>Login Screen</h1>
            <hr />
            <div className="my-2">
                <input type="text" placeholder="Username" onChange={handleInput} name="username" />
            </div>
            <button className="btn btn-primary" onClick={handleLogin}>
                Login
            </button>
        </div>
    );
};

export default LoginScreen;
