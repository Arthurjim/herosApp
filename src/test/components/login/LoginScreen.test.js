import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import LoginScreen from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockNavigate,
}));

describe("Pruebas en LoginScreen", () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false,
        },
    };
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={["/login"]}>
                <Routes>
                    <Route path="/login" element={<LoginScreen />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    );
    test("Debe de hacer match con el snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Debe de realizar el dispatch y la navegacion", () => {
        wrapper.find("button").prop("onClick")();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: { name: "Arturo" },
        });
        expect(mockNavigate).toHaveBeenCalledWith("/", { replace: true });

        localStorage.setItem("lastPath", "/dc");

        wrapper.find("button").prop("onClick")();
        expect(mockNavigate).toHaveBeenCalledWith("/dc", { replace: true });

    });
});
