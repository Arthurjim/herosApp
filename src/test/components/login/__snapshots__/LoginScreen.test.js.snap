// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen Debe de hacer match con el snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
